plugins/src/main/java/co/bound/codeartifact/CodeArtifactRepoProvider.java

```diff
@@ -19,15 +20,22 @@
 import java.util.Properties;
 
 public abstract class CodeArtifactRepoProvider implements BuildService<CodeArtifactRepoProvider.Params> {
+    private static final String CODEARTIFACT_REPOSITORY_NAME = "codeartifact";
```

Comment:
Can we make the repo name public, so we can use the field in build script?

Reply:
The tricky thing with this is exactly the examples in the added tests - there we want to declare repository filtering for plugin repositories. And plugin repositories are defined in the `pluginManagement` block, which needs to be the first thing in the buildscript and is the first thing that is evaluated by Gradle. At that point, we don't have the codeartifact plugin on the classpath and thus can not import and reference the classes from the plugin.

Other than that, the constant should be usable in project repositories - those are evaluated after the settings plugins are applied and at that point we will have the plugin on the classpath.